doit/dependency.py

```diff
@@ -693,6 +693,12 @@ def get_status(self, task, tasks_dict, get_log=False):
 class UptodateCalculator(object):
     """Base class for 'uptodate' that need access to all tasks
     """
+    
```

Comment:
Thanks. I think I understand the problem (not 100% sure).

The problem is that for Task I did not define `__getstate__` method. Because sometimes I want to really pass the full Task including the attributes that might contain functions, and sometimes I know I can leave these attributes out.

So the fix would be to define a `Task.__getstate__` that removes unsafe attributes by default, **and** figure out a way to create a pickle from an object ignoring its `__getstate__`! I am not sure how to do that. Maybe using cloudpickle will just solve the problem in a easier way.

I will take some time to experiment on that... meanwhile you can remove this change in `UptodateCalculator` (and let the test fail). This change will actually break the code if you have a `getargs` attribute in a DelayedTask (the tests doesnt cover that). So I will merge the improvement on not using wheels and fix this issue later. Of course you are welcome to try to fix it yourself :)